compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/BIRGen.java

```diff
@@ -2393,6 +2379,81 @@ public void visit(BLangRegExpTemplateLiteral regExpTemplateLiteral) {
         this.env.targetOperand = toVarRef;
     }
 
+    private BIRNonTerminator.NewStructure createNewStructureInst(List<BIRNode.BIRMappingConstructorEntry> fields,
+                                                                 BIROperand toVarRef, BType type, Location pos) {
+        if (getTypedescVariable(type) != null) {
+            return new BIRNonTerminator.NewStructure(pos, toVarRef, new BIROperand(getTypedescVariable(type)), fields);
+        } else {
+            createNewTypedescInst(type, type, pos);
+            return new BIRNonTerminator.NewStructure(pos, toVarRef, this.env.targetOperand, fields);
+        }
+    }
+
+    private BIRVariableDcl getTypedescVariable(BType type) {
+        for (Map.Entry<BSymbol, BIRVariableDcl> entry : env.symbolVarMap.entrySet()) {
+            if (isTypeDescSymbol(entry.getKey(), type)) {
+                return env.symbolVarMap.get(entry.getKey());
+            }
+        }
+
+        for (Map.Entry<BSymbol, BIRGlobalVariableDcl> entry : this.globalVarMap.entrySet()) {
+            if (isTypeDescSymbol(entry.getKey(), Types.getImpliedType(type))) {
```

Comment:
Can't we separate `Types.getImpliedType(type)` into a variable so that function is called only once.